routes.jsx

```diff
@@ -27,21 +27,25 @@ const App = React.createClass({
     };
   },
   componentDidMount() {
-    if (!this.state.suppressSplashScreen) {
+    if (this.refs.splash) {
       setTimeout(this.dismissSplash, this.props.dismissTimeout);
     }
   },
   dismissSplash() {
-    this.splash.classList.add(`dismissed`);
+    this.refs.splash.classList.add(`dismissed`);
     document.querySelector(`#app`).classList.add(`splash-dismissed`);
-    localstorage.setItem(`suppressSplashScreen`, `true`);
+    this.refs.splash.addEventListener(`transitionend`, () => {
+      // wait for CSS animation to finish first before we
+      // set `suppressSplashScreen` in localStorage to true
+      localstorage.setItem(`suppressSplashScreen`, `true`);
```

Comment:
not just localstorage, also in state, so let's also add a `this.setState({ suppressSplashScreen: true })` here to make sure render() `null`s the splash div